lib/kiwi_config.rb

```diff
@@ -124,7 +125,8 @@ def inject_extracted_files(output_location)
       FileUtils.mkdir_p(destination, mode: 01777)
       filter = "unmanaged_files_#{@name}_excludes"
 
-      @system_description.unmanaged_files.export_files_as_tarballs(destination)
+      @system_description.unmanaged_files.file_access.
+        export_files_as_tarballs(destination)
```

Comment:
This would not provide any separation between the two ways how files are stored, and there would not be any encapsulation by name space. The `self.is_a?` is a shortcut for now. This in the end should come from the scopes itself.

For me the most important step is to collect all the code accessing the files in one place. We then still can see what is the best way to make the API accessible.